app/main.py

```diff
@@ -2,24 +2,61 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: any, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: any, owner: any) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: any, value: int) -> None:
+        if not self.min_amount <= value <= self.max_amount:
+            raise ValueError
```

Comment:
In case of a `ValueError`, it would be helpful to provide a custom error message to indicate what the problem is. This can help with debugging and also provide a more meaningful error to the user.